### PR TITLE
Change Password API

### PR DESCRIPTION
**Pull request by hibearpanda**
This branch was branched off of the branch in #34 
## Changes
- Add API to change password

### CODE REVIEW

File: services/user.go [hunk 1]
```diff
@@ -183,6 +183,32 @@ func UpdateEmailConfirmed(id int, db *sql.DB) error {
 	return nil
 }
 
+// UpdatePasswordByID updates a user's password given a user's ID
+func UpdatePasswordByID(id int, p string, db *sql.DB) (models.User, error) {
```
Comment (minocys) [inline services/user.go]: This function should just return an error

File: services/user.go [hunk 2]
```diff
@@ -183,6 +183,32 @@ func UpdateEmailConfirmed(id int, db *sql.DB) error {
...
+	var u models.User
+
+	err := db.QueryRow(
+		"UPDATE user_account SET password = $1 WHERE id = $2 RETURNING *",
```
Comment (minocys) [inline services/user.go]: returning * is unnecessary

File: services/user.go [hunk 3]
```diff
@@ -183,6 +183,32 @@ func UpdateEmailConfirmed(id int, db *sql.DB) error {
...
+		&u.Email,
+		&u.FirstName,
+		&u.LastName,
+		&u.Password,
+		&u.RoleID,
+		&u.CreatedAt,
+		&u.EmailConfirmed,
+	)
+
+	if err != nil {
```
Comment (minocys) [inline services/user.go]: should just return err from QueryRow

**Comment by minocys** [comment]
Pending changes and rebase off previous pull request

File: services/user.go (hunk 2 above)
Comment (hibearpanda) [inline services/user.go]: Done

File: services/user.go (hunk 1 above)
Comment (hibearpanda) [inline services/user.go]: Done

File: services/user.go (hunk 3 above)
Comment (hibearpanda) [inline services/user.go]: Done

**Pull request merged**